### PR TITLE
Add Form Input With Button variant of field group

### LINKED ISSUE #270: Inline Form Button
https://share.goabstract.com/3ed50205-4c09-4b35-9522-559403f33410

screen shot 2018-09-17 at 2 16 56 pm

### PR DESCRIPTION
**Pull request by kevinmpowell**
I added a variant to the `form_field_group` component that allows this alternate layout of a label, input, and button.

Closes #270

### CODE REVIEW

File: library/components/form/form.scss [hunk 1]
```diff
@@ -99,6 +99,26 @@ $spirit-space-field-group-horizontal-margin:  0 16px 0 0;
...
+
+  .spirit-button {
+    bottom: 0;
+    margin: 0 $spirit-space-generic-2-x $spirit-space-generic-1-x 0;
+    position: absolute;
+    right: 0;
+  }
+
+  .spirit-form__input-field {
+    padding-right: 120px; // Add some padding inside the form input to accomodate the button width, will likely need overridden on a case-by-case basis
```
Comment (jamesmelzer) [inline library/components/form/form.scss]: This makes me a little nervous. Let's be sure to document it.

**Comment by joesnellpdx** [comment]
Good catch @lauralvictor 
@kevinmpowell Able to take a look? Line-height not being respected in ie. May need to explicitly declare height for input.

**Comment by kevinmpowell** [comment]
@joesnellpdx I'll revisit tomorrow.

**Comment by kevinmpowell** [comment]
Nice catch @lauralvictor and thanks for the direction @joesnellpdx. I've updated with a fix.

**Pull request merged**